tutorials/basic-topics/sorting/insertion-sort.md

```diff
@@ -11,4 +11,222 @@ keywords:
+* Take out the last two elements because those are the largest two elements in our array.
+* Put the values in the expression and return it.
+
+## Implementation
+
+Following are the implementation of the above mentioned problem:
+
+<Tabs>
+<TabItem value="cpp" label="C++">
+<SolutionAuthor name="@Shivashish-rwt"/>
+
+```cpp
+class Solution {
+public:
+    void insertionSort(vector<int> &arr, int size_of_array) {
+        for (int i = 1; i < size_of_array; i++) {
+            // Picking element from array
+            int nums = arr[i]; 
+            int j = i - 1;
+            
+            // Comparing nums with all the elements left to it
+            while (j >= 0 and arr[j] > nums) { 
+                // Shifting the greater element to the right
+                arr[j + 1] = arr[j]; 
+                j--;
+            }
+            // Placing selected element at correct position
+            arr[j + 1] = nums; 
+        }
+    }
+
+    int maxProduct(vector<int>& nums) {
+        insertionSort(nums,nums.size());
+        int n = nums.size();
+        return (nums[n - 1] - 1)*(nums[n - 2] - 1);
+    }
+};
+```
+
+</TabItem>
+
+<TabItem value="java" label="Java">
+<SolutionAuthor name="@Shivashish-rwt"/>
+
+```java
+class Solution {
+    void insertion_sort(int arr[],int size_of_array) {
+		for (int i = 1; i < size_of_array; ++i) {
+            // Picking element from array
+			int nums = arr[i]; 
+			int j = i - 1;
+
+            // Comparing nums with all the elements left to it
+			while (j >= 0 && arr[j] > nums) { 
+                // Shifting the greater element to the right
+				arr[j + 1] = arr[j]; 
+				j = j - 1;
+			}
+            // Placing selected element at correct position
+			arr[j + 1] = nums; 
+		}
+	}
+    public int maxProduct(int[] nums) {
+        insertion_sort(nums,nums.length);
+        int n = nums.length;
+        return (nums[n - 1] - 1)*(nums[n - 2] - 1);      
+    }
+}
+```
+
+</TabItem>
+
+<TabItem value="py" label="Python">
+<SolutionAuthor name="@Shivashish-rwt"/>
+
+```py
+class Solution:
+    def insertionSort(arr,size_of_array):
+        
+        for i in range(1, size_of_array):
+            # Picking element from array
+            nums = arr[i]  
+            j = i - 1
+              
+            # Comparing nums with all the elements left to it  
+            while j >= 0 and arr[j] > nums: 
+                # Shifting the greater element to the right
+                arr[j + 1] = arr[j] 
+                j = j - 1
+
+            # Placing selected element at correct position
+            arr[j + 1] = nums 
+        
+
+    def maxProduct(self, nums: List[int]) -> int:
+        Solution.insertionSort(nums,len(nums))
+        return (nums[-1] - 1) * (nums[-2] - 1)
+```
+
+</TabItem>
+
+</Tabs>
+
+## Time Complexity
+
+Apart from sorting the the given array we are not doing anything in the solution. The insertion sort would take $\mathcal{O}(n^2)$ to sort the array.
```

Comment:
Instead of you can use simple LaTex for the complexity

$O(n)$ - would be enough. Change it and run it locally and make changes.